packages/block-library/src/template-part/edit/index.js

```diff
@@ -17,89 +16,48 @@ import {
 } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { chevronUp, chevronDown } from '@wordpress/icons';
-import { serialize } from '@wordpress/blocks';
+
 /**
  * Internal dependencies
  */
-import useTemplatePartPost from './use-template-part-post';
 import TemplatePartNamePanel from './name-panel';
 import TemplatePartInnerBlocks from './inner-blocks';
 import TemplatePartPlaceholder from './placeholder';
 import TemplatePartSelection from './selection';
 
 export default function TemplatePartEdit( {
-	attributes: { postId: _postId, slug, theme, tagName: TagName = 'div' },
+	attributes: { slug, theme, tagName: TagName = 'div' },
 	setAttributes,
 	clientId,
 } ) {
-	const initialSlug = useRef( slug );
-	const initialTheme = useRef( theme );
-	const initialContent = useRef();
-
-	// Resolve the post ID if not set, and load its post.
-	const postId = useTemplatePartPost( _postId, slug, theme );
+	const templatePartId = theme + '|' + slug;
 
 	// Set the postId block attribute if it did not exist,
 	// but wait until the inner blocks have loaded to allow
 	// new edits to trigger this.
-	const { innerBlocks, expectedContent } = useSelect(
+	const { isResolved, innerBlocks } = useSelect(
 		( select ) => {
 			const { getBlocks } = select( 'core/block-editor' );
-			const entityRecord = select( 'core' ).getEntityRecord(
-				'postType',
-				'wp_template_part',
-				postId
-			);
+			const entityRecord =
+				theme && slug
+					? select( 'core' ).getEntityRecord(
+							'postType',
+							'wp_template_part',
+							theme + '|' + slug
+					  )
+					: null;
 
 			return {
 				innerBlocks: getBlocks( clientId ),
-				expectedContent: entityRecord?.content.raw,
+				isResolved: !! entityRecord,
 			};
 		},
-		[ clientId, postId ]
+		[ templatePartId, clientId ]
 	);
-	const { editEntityRecord } = useDispatch( 'core' );
-	useEffect( () => {
```

Comment:
It feels great to not have to rely on this effect anymore

Reply:
I think we still need to `setAttributes( { postId } )` at some point once the template part is edited.

Without this (currently testing this PR), if we edit a template part (add a block) and save the changes disappear after save and it reverts to the theme file.

Reply:
I explain it in the description of the PR. A couple `theme + slug` is unique regardless of whether the template is edited or not. 

So fist when theme loads, before you make any edit, this will resolve to the "file" version and as soon as you make edits to this template (or template part), it will resolve to the CPT version.

This  is the main premise of this PR. Stop relying on the "id" of the CPT post because it's something that is inconsistent depending on the current state of the template (edited or not)

Reply:
My main point is that editing a "file" version is broken in testing.  Please pardon/disregard my focus on postIds.  If I edit a file based version that is in the editor, all the changes disappear when I save.

Reply:
Oh I missed that part, sorry, I'll test and see what's the issue.

Reply:
> Oh I missed that part, sorry, I'll test and see what's the issue.

My bad, I could have been more 'to the point' with less of the excess postId blabber. 😆   Thanks!

Reply:
I think this is fixed now.

Reply:
Tested and confirmed.  Thanks Riad!